lang/cs/Org.Apache.REEF.Tests/Functional/FaultTolerant/TestResubmitContext.cs

```diff
@@ -0,0 +1,235 @@
+            ValidateMessageSuccessfullyLogged(messages1, "driver", DriverStdout, testFolder, 2);
+
+            CleanUp(testFolder);
+        }
+
+        public IConfiguration DriverConfigurations()
+        {
+            return DriverConfiguration.ConfigurationModule
+                .Set(DriverConfiguration.OnDriverStarted, GenericType<ResubmitEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnEvaluatorAllocated, GenericType<ResubmitEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnEvaluatorCompleted, GenericType<ResubmitEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnEvaluatorFailed, GenericType<ResubmitEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnContextActive, GenericType<ResubmitEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnTaskRunning, GenericType<ResubmitEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnContextFailed, GenericType<ResubmitEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnTaskCompleted, GenericType<ResubmitEvaluatorDriver>.Class)
+                .Set(DriverConfiguration.OnTaskFailed, GenericType<ResubmitEvaluatorDriver>.Class)
+                .Build();
+        }
+
+        private sealed class ResubmitEvaluatorDriver : 
+            IObserver<IDriverStarted>, 
+            IObserver<IAllocatedEvaluator>,
+            IObserver<ICompletedEvaluator>, 
+            IObserver<IFailedEvaluator>, 
+            IObserver<IRunningTask>,
+            IObserver<IFailedContext>,
+            IObserver<IFailedTask>, 
+            IObserver<IActiveContext>, 
+            IObserver<ICompletedTask>
+        {
+            private static readonly Logger Logger = Logger.GetLogger(typeof(ResubmitEvaluatorDriver));
+
+            private readonly IEvaluatorRequestor _requestor;
+            private int _taskNumber = 1;
+            private int _contextNumber = 1;
+
+            [Inject]
+            private ResubmitEvaluatorDriver(IEvaluatorRequestor requestor)
+            {
+                _requestor = requestor;
+            }
+
+            public void OnNext(IDriverStarted value)
+            {
+                _requestor.Submit(_requestor.NewBuilder().SetNumber(2).Build());
+            }
+
+            public void OnNext(IAllocatedEvaluator value)
+            {
+                Logger.Log(Level.Info, "AllocatedEvaluator: " + value.Id);
+                value.SubmitContext(
+                    ContextConfiguration.ConfigurationModule
+                        .Set(ContextConfiguration.Identifier, "ContextID" + _contextNumber++)
+                        .Build());
+            }
+
+            public void OnNext(IActiveContext value)
+            {
+                Logger.Log(Level.Info, "ActiveContext: " + value.Id);
+                value.SubmitTask(TaskConfiguration.ConfigurationModule
+                    .Set(TaskConfiguration.Identifier, TaskId + _taskNumber++)
```

Comment:
I don't like the use of the postfix increment operator here -- it's not as intuitive as it can be.